### PR TITLE
Single choice question added

### PR DESCRIPTION
**Pull request by Adam3004**
I am not proud of responsiveness of questionView. After submitting asnwers options becoming bigger. I do not really see what may cause this

### CODE REVIEW

File: src/pages/quiz/creation/Question.ts [hunk 1]
```diff
@@ -14,6 +14,21 @@ export type MultipleChoiceQuestion = {
     feedback: string[];
 };
 
+export type SingleChoiceQuestion = {
+    /** The textual content of the question. */
+    question: string;
+    /** The type of the question, indicating it's a "multiple-choice" type. */
```
Comment (krisx303) [inline src/pages/quiz/creation/Question.ts]: NIT: I guess the description could be changed or deleted

File: src/pages/quiz/creation/SingleChoiceQuestionView.tsx [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+                >
+                    <View
+                        style={[
+                            styles.checkBoxContainer
+                        ]}
+                        key={index}
+                    >
+                        <img
+                            style={{
+                                marginRight: !isEditable && userAnswer === index && question.answer === userAnswer ? 20 : 0,
```
Comment (krisx303) [inline src/pages/quiz/creation/SingleChoiceQuestionView.tsx]: I assume this works now, but I would define just two styles for this icon and use one condition to apply them:
```style={(!isEditable && userAnswer === index && question.answer === userAnswer)? styles.highlightedStyle : styles.defaultStyle}```

File: src/transport/HttpClient.ts [hunk 3]
```diff
@@ -99,6 +99,28 @@ class StubHttpClient implements HttpClientBase {
...
+                choices: [
+                    "1",
+                    "2",
+                    "3",
+                    "4",
+                    "5",
+                    "6",
+                ],
+                answer: 1,
+                feedback: [
```
Comment (krisx303) [inline src/transport/HttpClient.ts:116]: Really useful tips 😅

File: src/pages/quiz/creation/QuestionsScreen.tsx [hunk 4]
```diff
@@ -18,6 +17,8 @@ const getBaseUserAnswer = (q: Question): any => {
     switch (q.type) {
         case "multiple-choice":
             return Array(q.choices?.length || 0).fill(false);
+        case "single-choice":
```
Comment (krisx303) [inline src/pages/quiz/creation/QuestionsScreen.tsx:20]: I haven't tested this PR change yet, but does it mean the first answer is already checked as soon as you enter the question?

**Review by krisx303 — APPROVED** [review]
I don't have any major comments. Looks good for me

File: src/pages/quiz/creation/Question.ts (hunk 1 above)
Comment (Adam3004) [reply src/pages/quiz/creation/Question.ts]: I was wondered if it should be and imo it still may be usefull even in single choice

Comment (Adam3004) [reply src/pages/quiz/creation/Question.ts]: sry its late, corrected

File: src/pages/quiz/creation/SingleChoiceQuestionView.tsx (hunk 2 above)
Comment (Adam3004) [reply src/pages/quiz/creation/SingleChoiceQuestionView.tsx]: great idea

File: src/pages/quiz/creation/QuestionsScreen.tsx (hunk 4 above)
Comment (Adam3004) [reply src/pages/quiz/creation/QuestionsScreen.tsx:20]: yup

**Pull request merged**